jcl/src/jdk.attach/share/classes/com/ibm/tools/attach/attacher/OpenJ9AttachProvider.java

```diff
@@ -81,13 +86,30 @@ public OpenJ9VirtualMachine attachVirtualMachine (
 		}
 
 		OpenJ9VirtualMachine vm = new OpenJ9VirtualMachine(this, descriptor.id());
-		vm.attachTarget();
+		PrivilegedExceptionAction<Object> action = () -> {vm.attachTarget(); return null;};
+		try {
+			AccessController.doPrivileged(action);
+		} catch (PrivilegedActionException e) {
+			Throwable cause = e.getCause();
+			if (cause instanceof AttachNotSupportedException) {
+				throw (AttachNotSupportedException) cause;
+			} else if (cause instanceof IOException) {
+				throw (IOException) cause;
+			} else {
```

Comment:
There should also be a check for instances of RuntimeException or Error, as they can be re-thrown without a wrapper.